ts/client/src/client.ts

```diff
@@ -810,86 +811,47 @@ export class MangoClient {
       )) as AccountInfo<Buffer>,
     );
   }
 
-  private getMangoAccountFromAi(
+  private async getMangoAccountFromAi(
     mangoAccountPk: PublicKey,
     ai: AccountInfo<Buffer>,
-  ): MangoAccount {
+  ): Promise<MangoAccount> {
     const decodedMangoAccount = this.program.coder.accounts.decode(
       'mangoAccount',
       ai.data,
     );
 
-    // Start reading tokenConditionalSwaps from this point in account buffer
-    const start =
-      // anchor headers
-      8 +
-      // fixed portion
-      (this.program as any)._coder.types.typeLayouts.get('MangoAccountFixed')
-        .span +
-      // dynamic portion header
-      8 +
-      4 +
-      // tokens
-      4 +
-      decodedMangoAccount.tokens.length *
-        (this.program as any)._coder.types.typeLayouts.get('TokenPosition')
-          .span +
-      4 +
-      // serum3Orders
-      4 +
-      decodedMangoAccount.serum3.length *
-        (this.program as any)._coder.types.typeLayouts.get('Serum3Orders')
-          .span +
-      4 +
-      // perpPositions
-      4 +
-      decodedMangoAccount.perps.length *
-        (this.program as any)._coder.types.typeLayouts.get('PerpPosition')
-          .span +
-      4 +
-      // perpOpenOrders
-      4 +
-      decodedMangoAccount.perpOpenOrders.length *
-        (this.program as any)._coder.types.typeLayouts.get('PerpOpenOrder')
-          .span +
-      4 +
-      // tokenConditionalSwaps
-      4;
-
-    let tokenConditionalSwaps;
-    if (
-      // mango accounts which haven't been expanded for tokenConditionalSwaps
-      start ==
-      ai.data.length +
-        4 + // padding
-        4 // length of tokenConditionalSwaps
-    ) {
-      tokenConditionalSwaps = [];
-    } else {
-      // mango accounts which have been expanded for tokenConditionalSwaps already
-      const tcsSize = (this.program as any)._coder.types.typeLayouts.get(
-        'TokenConditionalSwap',
-      ).span;
-      const tcsSequenceLength = u32.u32().decode(ai?.data, start - 4);
-      tokenConditionalSwaps = range(tcsSequenceLength).map((i) =>
-        (this.program as any)._coder.types.typeLayouts
-          .get('TokenConditionalSwap')
-          .decode(
-            ai.data.slice(start + i * tcsSize, start + (i + 1) * tcsSize),
-          ),
-      );
-      if (ai.data.length - start - tcsSequenceLength * tcsSize !== 0) {
-        throw new Error("The size of the account doesn't match expectations");
-      }
-    }
+    // Re-encode decoded mango account with v1 layout, this will help identifying
+    // if account is of type v1 or v2
+    // Do whole encoding manually, since anchor uses a buffer of a constant length which is too small
+    let mangoAccountV1Buffer = Buffer.alloc(ai.data.length);
+    const layout =
+      this.program.coder.accounts['accountLayouts'].get('mangoAccount');
+    const len = layout.encode(decodedMangoAccount, mangoAccountV1Buffer);
+    mangoAccountV1Buffer = mangoAccountV1Buffer.subarray(0, len);
```

Comment:
Since the len is all we care about, we could skip getting the buffer and just compare to len + 8